src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,103 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static final int SIZE_ARRAY = 10;
+    private static final double GROW_FACTOR = 1.5;
+    private T[] array;
+    private int size;
+
+    public ArrayList() {
+        array = (T[])new Object[SIZE_ARRAY];
+    }
+
     @Override
     public void add(T value) {
-
+        if (size == array.length) {
+            growArray();
+        }
+        array[size++] = value;
     }
 
     @Override
     public void add(T value, int index) {
-
+        checkIndexOutOfBoundsAdd(index);
```

Comment:
Here, you can add a check using `if (index == size)` and call the `add(value)` method in such cases. 
Then, you can call `checkIndexOutOfBounds` method. This way, you can remove `checkIndexOutOfBoundsAdd` method.

Reply:
as Uliana said here you can use your checkIndexOutOfBounds method as case when index==size already checked above
and you can remove checkIndexOutOfBoundsAdd method as it's no longer required